File: java/src/main/java/problems201_300/MoveZeroes.java
package problems201_300;

enum MoveZeroes {
    DEFAULT {
        @Override
        void moveZeroes(int[] nums) {
            int i = 0;
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[i] == 0 && nums[j] != 0) {
                    int temp = nums[i];
                    nums[i] = nums[j];
                    nums[j] = temp;
                    i++;
                } else if (nums[i] != 0) {
                    i++;
                }
            }
        }
    };

    abstract void moveZeroes(int[] nums);
}

File: kotlin/src/main/kotlin/problems201_300/MoveZeroes.kt
package problems201_300

enum class MoveZeroes {
    DEFAULT {
        override fun moveZeroes(nums: IntArray) {
            var i = 0
            for (j in i + 1 until nums.size) {
                if (nums[i] == 0 && nums[j] != 0) {
                    val temp = nums[i]
                    nums[i] = nums[j]
                    nums[j] = temp
                    i++
                } else if (nums[i] != 0) {
                    i++
                }
            }
        }
    },
    ;

    abstract fun moveZeroes(nums: IntArray)
}
